test/subcommand_flags.cc

```diff
@@ -0,0 +1,314 @@
+#include <array>
+
+#include "gtest/gtest.h"
+#include "stout/flags.h"
+#include "test/test.pb.h"
+
+TEST(FlagsTest, MissingSubcommandExtension) {
+  EXPECT_DEATH(
+      []() {
+        test::SubcommandFlagsWithoutExtension flag;
+        auto builder = stout::flags::Parser::Builder(&flag);
+        builder.Build();
+      }(),
+      "Every field of the 'oneof subcommand' must have"
+      " .stout.v1.subcommand. extension");
+}
+
+TEST(FlagsTest, IncorrectSubcommandExtension) {
+  EXPECT_DEATH(
+      []() {
+        test::FlagsWithIncorrectExtension flag;
+        auto builder = stout::flags::Parser::Builder(&flag);
+        builder.Build();
+      }(),
+      ".stout.v1.subcommand. extension should be inside only"
+      " a 'oneof subcommand' field");
+}
+
+TEST(FlagsTest, IncorrectOneofName) {
+  EXPECT_DEATH(
+      []() {
+        test::IncorrectOneofName flag;
+        auto builder = stout::flags::Parser::Builder(&flag);
+        builder.Build();
+      }(),
+      "'oneof' field must have 'subcommand' name. "
+      "Other names are illegal");
+}
+
+TEST(FlagsTest, SubcommandFlagExtension) {
+  EXPECT_DEATH(
+      []() {
+        test::SubcommandFlagExtension flag;
+        auto builder = stout::flags::Parser::Builder(&flag);
+        builder.Build();
+      }(),
+      "Every field of the 'oneof subcommand' must have"
+      " .stout.v1.subcommand. extension");
+}
+
+TEST(FlagsTest, SimpleSubcommandBuildSucceed) {
+  test::SimpleSubcommandSucceed flags;
+
+  auto parser = stout::flags::Parser::Builder(&flags).Build();
+
+  std::array arguments = {
+      "program",
+      "--b",
+      "build",
+      "--other_flag=13",
+  };
+
+  int argc = arguments.size();
+  const char** argv = arguments.data();
+
+  parser.Parse(&argc, &argv);
+
+  EXPECT_EQ(1, argc);
+  EXPECT_EQ("program", argv[0]);
+  EXPECT_TRUE(flags.b());
+  EXPECT_EQ(13, flags.build().other_flag());
```

Comment:
Let's first `ASSERT()` that `has_build()`! Here and everywhere else! It's an assert because if you don't have `build()` then there is no reason to check anything else in the test.